### ISSUE TITLE
Certificate/provisioning profile mismatch despite no changes

### ISSUE THREAD

**Issue by CodeWithOz** [fastlane/fastlane#15583]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

I started getting the error: `Provisioning profile "match AppStore [app-id]" doesn't include signing certificate "iPhone Developer: [name] ([certificate ID])". (in target '[target]')”` despite changing nothing related to Fastlane build. 

Basically, 2 days ago I used Fastlane with the `beta` lane (see below) and it worked fine, then today it gave the above error. In between both days, I upgraded 2 cordova plugins, but I don't see how they would cause Xcode to begin to mismatch certificates. The last time I got this error (about a week ago while setting up Fastlane), it was because in the `beta` lane I had set the `type` for the cordova action to "development" whereas I set the `type` for `match` to "appstore". I fixed that by setting both of those values to "appstore". I haven't made any changes to Fastlane or my certificates/profiles since it last worked, yet now I'm getting an error. I've checked Apple Developer Portal and the certificates are still valid there. And the private keys for my certificates are all present in Keychain Access. Really stumped about what could be causing this. The `simulate` lane works by the way.

##### Expected behavior
The build would complete successfully as it has been doing over the past week, given that the certificates and profiles haven't changed since they last worked.

##### Complete output when running fastlane, including the stack trace and command used

<details>
  <pre>ERROR [2019-11-02 12:32:14.08]: Exit status of command 'cordova compile ios --release --device --packageType=app-store --developmentTeam=[ID HERE] --provisioningProfile=[PROFILE ID HERE] -- ' was 65 instead of 0.
Reading build config file:
Building project: /Users/[USER]/Documents/[FOLDER]/platforms/ios/[APP-NAME].xcworkspace
        Configuration: Release
        Platform: device
        Target:
Running command: xcodebuild -workspace [APP-NAME].xcworkspace -scheme [SCHEME] -configuration Release -destination generic/platform=iOS -archivePath [APP-NAME].xcarchive archive CONFIGURATION_BUILD_DIR=/Users/[USER]/Documents/[FOLDER]/platforms/ios/build/device SHARED_PRECOMPS_DIR=/Users/[USER]/Documents/[FOLDER]/platforms/ios/build/sharedpch
User defaults from command line:
    IDEArchivePathOverride = /Users/[USER]/Documents/[FOLDER]/platforms/ios/[APP-NAME].xcarchive

Build settings from command line:
    CONFIGURATION_BUILD_DIR = /Users/[USER]/Documents/[FOLDER]/platforms/ios/build/device
    SHARED_PRECOMPS_DIR = /Users/[USER]/Documents/[FOLDER]/platforms/ios/build/sharedpch

note: Using new build system
note: Planning build
note: Constructing build description
Build system information
error: Provisioning profile "match AppStore [APP-ID]" doesn't include signing certificate "iPhone Developer: [NAME AND ID]". (in target '[TARGET]')

** ARCHIVE FAILED **

xcodebuild: Command failed with exit code 65</pre>
</details>

### Environment

<details>
  <pre><details><summary>🚫 fastlane environment 🚫</summary>

### Stack

| Key                         | Value                                       |
| --------------------------- | ------------------------------------------- |
| OS                          | 10.13.6                                     |
| Ruby                        | 2.6.3                                       |
| Bundler?                    | true                                        |
| Git                         | git version 2.17.2 (Apple Git-113)          |
| Installation Source         | ~/.rvm/rubies/ruby-2.6.3/bin/fastlane       |
| Host                        | Mac OS X 10.13.6 (17G65)                    |
| Ruby Lib Dir                | ~/.rvm/rubies/ruby-2.6.3/lib                |
| OpenSSL Version             | OpenSSL 1.1.1d  10 Sep 2019                 |
| Is contained                | false                                       |
| Is homebrew                 | false                                       |
| Is installed via Fabric.app | false                                       |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/ |
| Xcode Version               | 10.1                                        |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   |             |   |
| LANGUAGE |             |   |

### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
platform :ios do
  desc "Manage ios builds"

  lane :simulate do
    match(type: "development")
    cordova(
      platform: 'ios',
      type: 'development',
      release: false,
      device: false,
      build_number: get_build_number_from_plist(
        xcodeproj: './platforms/ios/[APP-NAME].xcodeproj'
      )
    )
  end

  lane :beta do
    desc "Create a build to upload to TestFlight"

    match(type: "appstore")
    increment_build_number_in_plist(
      xcodeproj: './platforms/ios/[APP-NAME].xcodeproj'
    )
    cordova(
      platform: 'ios',
      build_number: get_build_number_from_plist(
        xcodeproj: './platforms/ios/[APP-NAME].xcodeproj'
      )
    )
    upload_to_testflight(
      ipa: "./platforms/ios/build/device/[APP-NAME].ipa",
      skip_waiting_for_build_processing: true
    )
  end
end
```
</details>

<details><summary>`./fastlane/Appfile`</summary>

```ruby
# View this URL if you are a part of multiple teams in the Apple Developer Portal
# https://docs.fastlane.tools/advanced/Appfile/

app_identifier "[APP ID HERE]"

# TODO: Enter your Apple Developer Portal username
apple_id "[APPLE ID HERE]"

team_id "[TEAM ID HERE]"
```
</details>

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.134.0 | ✅ Up-To-Date  |

### Loaded fastlane plugins:

| Plugin                     | Version | Update-Status      |
| -------------------------- | ------- | ------------------ |
| fastlane-plugin-cordova    | 2.2.1   | ✅ Up-To-Date       |
| fastlane-plugin-versioning | 0.3.4   | 🚫 Update available |

<details><summary><b>Loaded gems</b></summary>

| Gem                        | Version      |
| -------------------------- | ------------ |
| did_you_mean               | 1.3.0        |
| executable-hooks           | 1.6.0        |
| bundler-unload             | 1.0.2        |
| rubygems-bundler           | 1.4.5        |
| bundler                    | 1.17.3       |
| etc                        | 1.0.1        |
| forwardable                | 1.2.0        |
| CFPropertyList             | 3.0.1        |
| public_suffix              | 2.0.5        |
| addressable                | 2.7.0        |
| atomos                     | 0.1.3        |
| babosa                     | 1.0.3        |
| claide                     | 1.0.3        |
| colored                    | 1.2          |
| colored2                   | 3.1.2        |
| highline                   | 1.7.10       |
| commander-fastlane         | 4.4.6        |
| declarative                | 0.0.10       |
| declarative-option         | 0.1.0        |
| digest-crc                 | 0.4.1        |
| unf_ext                    | 0.0.7.6      |
| unf                        | 0.1.4        |
| domain_name                | 0.5.20190701 |
| dotenv                     | 2.7.5        |
| emoji_regex                | 1.0.1        |
| excon                      | 0.67.0       |
| multipart-post             | 2.0.0        |
| faraday                    | 0.17.0       |
| http-cookie                | 1.0.3        |
| faraday-cookie_jar         | 0.0.6        |
| faraday_middleware         | 0.13.1       |
| fastimage                  | 2.1.7        |
| gh_inspector               | 1.1.3        |
| jwt                        | 2.1.0        |
| memoist                    | 0.16.0       |
| multi_json                 | 1.14.1       |
| os                         | 1.0.1        |
| signet                     | 0.12.0       |
| googleauth                 | 0.6.7        |
| httpclient                 | 2.8.3        |
| mime-types-data            | 3.2019.1009  |
| mime-types                 | 3.3          |
| uber                       | 0.1.0        |
| representable              | 3.0.4        |
| retriable                  | 3.1.2        |
| google-api-client          | 0.23.9       |
| google-cloud-env           | 1.3.0        |
| google-cloud-core          | 1.3.2        |
| google-cloud-storage       | 1.16.0       |
| json                       | 2.2.0        |
| mini_magick                | 4.9.5        |
| multi_xml                  | 0.6.0        |
| plist                      | 3.5.0        |
| rubyzip                    | 1.3.0        |
| security                   | 0.1.3        |
| naturally                  | 2.2.0        |
| simctl                     | 1.6.6        |
| slack-notifier             | 2.3.2        |
| terminal-notifier          | 2.0.0        |
| unicode-display_width      | 1.6.0        |
| terminal-table             | 1.8.0        |
| tty-screen                 | 0.7.0        |
| tty-cursor                 | 0.7.0        |
| tty-spinner                | 0.9.1        |
| word_wrap                  | 1.0.0        |
| nanaimo                    | 0.2.6        |
| xcodeproj                  | 1.13.0       |
| rouge                      | 2.0.7        |
| xcpretty                   | 0.3.0        |
| xcpretty-travis-formatter  | 1.0.0        |
| fastlane-plugin-cordova    | 2.2.1        |
| fastlane-plugin-versioning | 0.3.4        |
</details>

*generated on:* **2019-11-03**
</details></pre>
</details>

**Comment by RenGate** [comment]
I fixed this problem by removing all development certificates from Keychain and running match to install only certificates which are used with provisioning profile.
Upon further investigation I found out that the issue was caused by new universal certificate (Xcode 11+) that was most likely created by Xcode. The new certificate had exactly the same identifier (M6UKPGV***) as old one that was stored in match repo. Due to the use of same identifiers Xcode tried to use wrong certificate.

**Comment by CodeWithOz** [comment]
@RenGate hmm that's very interesting. There's no such certificate in either my Keychain (and I've specifically deleted all extra/unused ones) or in my match repo. Do you have an idea of the specific step or Fastlane command that created the universal certificate?

**Comment by RenGate** [comment]
@CodeWithOz I doubt that it was created by Fastlane, in any case I didn't look further in this. Another thing worth trying is to remove all local provisioning profiles and run match again.

**Comment by CodeWithOz** [comment]
@RenGate yeah I tried that already, no luck 😞 . Really weird problem.

**Comment by andycwk** [comment]
Hay guys, so... I've been having the same issue (and banging head on wall for a week or so)

I've now pined down the culprit, at least for my use case, but writing it up here incase it helps anyone else in the same bucket of crud!

Basically I have an Ionic 3 build that was working until I updated cordova-ios (5.0.1) to the latest version (may have also updated branch-cordova-sdk (version 3.2.0) but can't remember of the top of my head)

Any way, it turns out there is a known issue with this issue of the branch SDK 
https://github.com/BranchMetrics/cordova-ionic-phonegap-branch-deep-linking-attribution/pull/577/files

While this code has not been merged, I've been able to identify it as a fix locally by modifying my `node-package` scripts.

As a fix for now.. I have stored the modified version of the `enableEntitlements.js` script and add the following in my beta fastlane to `hack` around the issue until something better is in place!

```
FileUtils.cp_r '../resources/branh-enableEntitlements.js', '../node_modules/branch-cordova-sdk/src/scripts/ios/enableEntitlements.js'
FileUtils.remove_dir '../platforms', true
FileUtils.remove_dir '../plugins', true
sh "(cd .. && ionic cordova build ios -- --buildConfig='#{ENV['BITRISE_CORDOVA_BUILD_CONFIGURATION']}' --buildFlag='-UseModernBuildSystem=0' --device --release --verbose)"

**Comment by CodeWithOz** [comment]
Forgot to respond but @andycwk thanks a lot for your suggestion! That link you shared actually set me on my way. I should add that while the solution in the sibling PR (["Option 2"](https://github.com/BranchMetrics/cordova-ionic-phonegap-branch-deep-linking-attribution/pull/578/files)) ultimately allowed me to get past the error, I was still getting a `"missing aps-environment"` error. The solution to that was [this comment](https://github.com/BranchMetrics/cordova-ionic-phonegap-branch-deep-linking-attribution/issues/465#issuecomment-402930496); I first added the script as a `before_prepare` hook (because I'm using Cordova instead of IonicPro) but it appeared to not be working. So I ran it manually in the terminal and that did the trick! Hope this helps someone else.
